backend-service/app/models/daos/DatasetDao.java

```diff
@@ -39,7 +40,9 @@
 public class DatasetDao {
 
   public static final String GET_DATASET_BY_ID = "SELECT * FROM dict_dataset WHERE id = :id";
+  public static final String GET_DATASET_BY_NAME = "SELECT * FROM dict_dataset WHERE name = :name";
```

Comment:
[name] is not very unique, so this is going to return multiple records in some cases.